### PR TITLE
cmd/snap-confine: mount support cleanups

### PR DESCRIPTION
**Pull request by mardy**
Here are a few cleanups, prompted by the new PR about a WSL2 mount-support module (https://github.com/snapcore/snapd/pull/11785).

There are just 4 small commits, but reviewing them as a whole can be a bit messy. I recommend have a look at each commit one by one to see the individual changes and the rationale.

### CODE REVIEW

File: cmd/snap-confine/mount-support-nvidia.c [hunk 1]
```diff
@@ -583,8 +579,8 @@ void sc_mount_nvidia_driver(const char *rootfs_dir, const char *base_snap_name)
 	}
 
 	sc_identity old = sc_set_effective_identity(sc_root_group_identity());
-	int res = mkdir(SC_LIB, 0755);
-	if (res != 0 && errno != EEXIST) {
+	int res = sc_nonfatal_mkpath(SC_LIB, 0755);
```
Comment (bboozzoo) [inline cmd/snap-confine/mount-support-nvidia.c]: if the parent of /var/lib/snapd/lib does not exist then there's lot more other problems already, hence the original code created the leaf path only, but overall the change is ok

**Review by bboozzoo — APPROVED** [review]
LGTM

**Review by stolowski — APPROVED** [review]
LGTM

**Comment by mardy** [comment]
@bboozzoo, @stolowski: sorry for force-pushing after your approval :-)

I realized that one commit was going against the direction where we are trying to move to: it was removing calls to chown(), under the reason that these calls are not needed since the directories are being created as root. Since we are trying to move into a future where snap-confine is no longer running as root, it's better to keep chown() around.

Reverting the comment would have made a bit too much noise in the history, so I just rebased and excluded it. :-)

**Review by alexmurray — APPROVED** [review]
LGTM!

**Comment by bboozzoo** [comment]
@mardy the branch fails to build, there's some places where SC_LIB was used, that weren't updated. Can you take a look?

**Pull request merged**